sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -401,9 +462,15 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
     return PI_INVALID_VALUE;
   }
 
+  const std::lock_guard<sycl::detail::SpinLock> Lock{*PiPlatformCacheMutex};
```

Comment:
This is not safe to access this var here as it might not be constructed yet. Hopefully it should be solved when all global variables are allocated(`new`) in `piInitialize` and deallocated in `piTearDown`.

Reply:
@romanovvlad, I created an issue ticket and assigned it to me in order to keep track of this issue.

https://github.com/intel/llvm/issues/5599